lib/proxy/log.rb

```diff
@@ -88,7 +88,11 @@ def call(env)
       logger.trace do
         if env['rack.input'] && !(body = env['rack.input'].read).empty?
           env['rack.input'].rewind
-          "Body: #{body}"
+          if body.size < 2000
```

Comment:
I'm wondering if we should also look at the `env['HTTP_CONTENT_TYPE']` to see if it's gzip/bzip2. Even small binary data is probably not very readable.

I'm also a fan of using constants for these values. `TRACE_MAX_BODY_SIZE` perhaps?

Reply:
Good ideas, implemented both. Actually more strict content type filtering - only JSON for now.